File: solutions.digamma.damas.jcr/src/main/java/solutions/digamma/damas/jcr/names/Namespace.java
package solutions.digamma.damas.jcr.names;

/**
 * Namespace constants and methods.
 *
 * @author Ahmad Shahwan
 */
public interface Namespace {

    /**
     * TypeNamespace delimiter.
     */
    String SEPARATOR = ":";

    /**
     * DMS name space.
     */
    String NAMESPACE = "dms";

    /**
     * Prefix name with namespace.
     *
     * @param name
     * @return
     */
    static String prefix(String name) {
        return String.format("%s%s%s", NAMESPACE, SEPARATOR, name);
    }
}

File: solutions.digamma.damas.jcr/src/main/kotlin/solutions/digamma/damas/jcr/names/Namespace.kt
package solutions.digamma.damas.jcr.names

/**
 * Namespace constants and methods.
 *
 * @author Ahmad Shahwan
 */
internal object Namespace {

    /**
     * TypeNamespace delimiter.
     */
    private const val SEPARATOR = ":"

    /**
     * DMS name space.
     */
    private const val NAMESPACE = "dms"

    /**
     * Prefix name with namespace.
     *
     * @param name
     * @return
     */
    fun prefix(name: String): String {
        return "%s%s%s".format(NAMESPACE, SEPARATOR, name)
    }
}
